### PR TITLE
Bugfix, use alcGetString instead of alGetString

### PR DESCRIPTION
**Pull request by DianeLooney**
Will allow for correct device names. Should not affect functionality.

## Context

Three important excepts from OpenAL's documentation:

1. `alGetString` does not support the parameters currently passed to it:
```
alGetString
Description
This function retrieves an OpenAL string property.
const ALchar * alGetString(
 ALenum param
);
Parameters
param The property to be returned
 AL_VENDOR
 AL_VERSION
 AL_RENDERER
 AL_EXTENSIONS 
```
2. `alcGetString` supports what we're trying to use `alGetString` to do:
```
To get the default device name, pass in NULL and
ALC_DEFAULT_DEVICE_SPECIFIER . 
```
3. `alcOpenDevice` was handling this error case correctly, silently. Since the return value from `alGetString` is `Null`, `alcOpenDevice` was functioning as expected, opening the default device, though not through the intended code-path.
```
The function call to open a device, alcOpenDevice, takes a string as input. The string should
contain either the name of a valid OpenAL rendering device, or NULL to request the default
device. 
```

So, in summary, `oto` was using the incorrect C functions, the particular null-values returned allowed this to go unnoticed silently, with the only side-effect being that the name of the device was not saved.

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Hi, 

It looks like you are correct. I was wondering why the current implementation worked...

LGTM, thank you for your contribution!
